### ISSUE TITLE
png not loading

### ISSUE THREAD

**Issue by dnlzm** [vipzhicheng/logseq-plugin-mark-map#25]
Any image.pnd embedded in a page will trigger the white square popup with spinning circle, but does not continue to showing the image.

**Comment by vipzhicheng** [comment]
I can not reproduce this issue, which means on my local(Mac), png image can be shown correctly, so what is your env?

**Comment by tws14er** [comment]
I am experienceing the same problem as dnlzm - image links just show a square popub with spinning circcle - no image shows. Windows 11 laptop. LogSeq version 0.7.9.
![Screenshot 2022-08-02 150438](https://user-images.githubusercontent.com/110497238/182473105-52de080e-04b3-405b-b606-8b76c2400303.png)

**Comment by working-name** [comment]
@vipzhicheng Same thing happens here, I was able to make it load by prepending the href's value for the lightbox link with `assets://`. Right now the path passed to that href is just `/Users/...` which is highly unlikely to load. Prepending with `file://` will not load either due to electron or whatever.

The other question I have - why not simply display the picture inline as well? I understand that it'll mess with the distance between branches so that may be difficult to handle, but with a modest height it should be achievable and not look terrible. Just thinking out loud.

**Comment by vipzhicheng** [comment]
My test case is that images are always in assets folder whether copy-n-paste or drag-n-drop with a relative path, so I'm not sure how it can be `/Users/`. And when I manually make it absolute path, I found that issues come up not only not loading but also messed up. So for now I can just suggest not using the absolute path style.

**Comment by working-name** [comment]
I don't manually edit the path, and did not change anything about Logseq or your plugin. It also has position issues on first initialization. Here's a recording of a vanilla logseq + markmap plugin install: [video](https://working.name/markmap-hiccups.mp4)

![video](https://working.name/markmap-hiccups.mp4)

**Comment by vipzhicheng** [comment]
@working-name  thx for your screencast, I think I know what happened. And I just release a new version which add `assets://` automatically. Please take a look if it works for you. And I also need to find a Windows machine to have a test.

**Comment by tws14er** [comment]
I checked the update on a Windows machine, and it works fine.

Thanks.

Tom

**Comment by working-name** [comment]
> @working-name thx for your screencast, I think I know what happened. And I just release a new version which add `assets://` automatically. Please take a look if it works for you. And I also need to find a Windows machine to have a test.

Confirmed, works great on macOS too. Thank you! 👍

**Comment by balta2ar** [comment]
I'm on Linux (Manjaro), using v0.2.11. I can't seem to display images. Let's say I have this structure with relative path to assets:

![image](https://user-images.githubusercontent.com/340676/197403492-d0e62ead-1205-4ea6-8db3-064f69b35dd0.png)

When I open Markmap, I see:
![image](https://user-images.githubusercontent.com/340676/197403549-35b3d7b7-45e5-445d-bec2-2f5236806b59.png)

And then if I click on the PNG link, I get:
![image](https://user-images.githubusercontent.com/340676/197403568-149f47c4-d793-4cc8-a5d0-3859a40c7543.png)

At the same time I have few errors in the log, but I don't know if they are related to this problem:
```
index.d33125f7.js:148 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'resetTheme')
    at J.<anonymous> (index.d33125f7.js:148:1408)
    at m.emit (logseq.0b585b9b.js:3:1808)
    at Ne.<anonymous> (logseq.0b585b9b.js:12:30151)
    at m.emit (logseq.0b585b9b.js:3:1768)
    at #lspmsg#settings# (logseq.0b585b9b.js:12:20841)
    at logseq.0b585b9b.js:12:16617

core.cljs:4051 [frontend.modules.shortcut.core] {:shortcut/register-shortcut {:id :plugin.logseq-move-block/move-block, :binding "ctrl+shift+m", :error #object[Error Error: Keyboard shortcut conflicts with existing shortcut: null]}, :line 70}

module.js:215 Unexpected error when checking capturing opt-out status: TypeError: Cannot read properties of undefined (reading 'token')
(anonymous) @ module.js:215
(anonymous) @ core.cljs:32
$APP.k.F @ core.cljs:11369
e @ handler.cljs:46
f @ handler.cljs:74
pr @ async.cljs:24
(anonymous) @ async.cljs:24
(anonymous) @ channels.cljs:30
hua @ channels.cljs:17
Iba.b.port1.onmessage @ promise.js:817
core.cljs:3669 TypeError: Cannot read properties of undefined (reading 'incr')
    at sc.<anonymous> (module.js:5334:7)
    at sc.capture (module.js:241:7)
    at core.cljs:32:5
    at $APP.k.F (core.cljs:11369:26)
    at e (handler.cljs:46:9)
    at Function.f [as F] (handler.cljs:74:5)
    at pr (async.cljs:24:4)
    at async.cljs:24:4
    at channels.cljs:30:1
    at hua (channels.cljs:17:3)
(anonymous) @ core.cljs:3669
$APP.k.F @ core.cljs:11369
e @ handler.cljs:46
f @ handler.cljs:74
pr @ async.cljs:24
(anonymous) @ async.cljs:24
(anonymous) @ channels.cljs:30
hua @ channels.cljs:17
Iba.b.port1.onmessage @ promise.js:817
module.js:215 Unexpected error when checking capturing opt-out status: TypeError: Cannot read properties of undefined (reading 'token')
(anonymous) @ module.js:215
(anonymous) @ core.cljs:32
$APP.k.F @ core.cljs:11369
e @ handler.cljs:46
f @ handler.cljs:74
pr @ async.cljs:24
(anonymous) @ async.cljs:24
(anonymous) @ channels.cljs:30
hua @ channels.cljs:17
Iba.b.port1.onmessage @ promise.js:817
core.cljs:3669 TypeError: Cannot read properties of undefined (reading 'incr')
    at sc.<anonymous> (module.js:5334:7)
    at sc.capture (module.js:241:7)
    at core.cljs:32:5
    at $APP.k.F (core.cljs:11369:26)
    at e (handler.cljs:46:9)
    at Function.f [as F] (handler.cljs:74:5)
    at pr (async.cljs:24:4)
    at async.cljs:24:4
    at channels.cljs:30:1
    at hua (channels.cljs:17:3)
```

**Comment by balta2ar** [comment]
I tried disabling all other plugins and leaving only this one active, and the problem remained.

**Comment by vipzhicheng** [comment]
> I'm on Linux (Manjaro), using v0.2.11. I can't seem to display images. Let's say I have this structure with relative path to assets:
> 
> ![image](https://user-images.githubusercontent.com/340676/197403492-d0e62ead-1205-4ea6-8db3-064f69b35dd0.png)

Seems you not use images correctly. You should make the images shown on Logseq first by the syntax `![image name](../assets/...)`

Logseq console always has some errors, just ignore them.

**Comment by balta2ar** [comment]
Interesting! I get this string when I copy an image to a clipboard from another app and I Ctrl-V it in logseq, and then logseq renders it correctly:
![image](https://user-images.githubusercontent.com/340676/197457603-bb5d6832-5ecb-48c4-ae30-9ec026daf1f0.png)
Looks like logseq can render both `![image name](../assets/...)` and `((../assets/...))`.

However, when I tried converting the first image link for the format your suggested:
```
- Ebbinghaus forgetting curve  
	- uten repetisjon  
		- ![name](../assets/image_1666542413704_0.png)  
	- med pepetisjon  
		- [[../assets/image_1666543027135_0.png]]  
```
I got this:
![image](https://user-images.githubusercontent.com/340676/197458917-d0ed4297-441e-4ad3-8ed5-50bbe89f7e45.png)
I don't know if it matters, but I'm using "org" format in this document, not "md".
